apps/opik-backend/src/main/java/com/comet/opik/domain/LlmProviderApiKeyService.java

```diff
@@ -123,6 +131,32 @@ public void updateApiKey(@NonNull UUID id, @NonNull ProviderApiKeyUpdate provide
         });
     }
 
+    @Override
+    public void delete(UUID id, String workspaceId) {
+        template.inTransaction(WRITE, handle -> {
+
+            var repository = handle.attach(LlmProviderApiKeyDAO.class);
+
+            repository.delete(id, workspaceId);
+
+            // Void return
+            return null;
+        });
```

Comment:
Minor: this method could just overload the other one and delegate the call to it after wrapping `id` argument in a `Set`. 

This way, our code is more maintainable.

Reply:
deleted